### PR TITLE
Print message when removing plugin with config

### PR DESCRIPTION
**Pull request by jasontedor**
When removing a plugin with a config directory, we preserve the config
directory. This is because the workflow for upgrading a plugin involves
removing and then installing the plugin again and losing the plugin
config in this case would be terrible. This commit causes a message
regarding this to be printed in case the user wants to manually delete
these files.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java [hunk 1]
```diff
@@ -81,10 +83,17 @@ void execute(Terminal terminal, String pluginName, Map<String, String> settings)
...
         Files.move(pluginDir, tmpPluginDir, StandardCopyOption.ATOMIC_MOVE);
         pluginPaths.add(tmpPluginDir);
 
         IOUtils.rm(pluginPaths.toArray(new Path[pluginPaths.size()]));
+
+        // we preserve the config files in case the user is upgrading the plugin, but we print
+        // a message so the user knows in case they want to remove manually
+        final Path pluginConfigDir = env.configFile().resolve(pluginName);
+        if (Files.exists(pluginConfigDir)) {
+            terminal.println("-> Preserving plugin config files: " + pluginConfigDir);
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java]: should we say something about upgrades in the message? I suspect people with "uninstall" in their head will have a WTF moment.  Probably good to say "delete manually if not needed".

**Comment by bleskes** [comment]
LGTM. Thanks for tests. Left a minor suggestion for the comment.

File: core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java]: I pushed 817bbfb2d118c4ccd28c9737c5763317979f6841.

**Comment by jasontedor** [comment]
retest this please

**Pull request merged**